tests/lustre/typechecking/shouldFail/test_circular_contract_eqns.lus

```diff
@@ -0,0 +1,8 @@
+contract topSpec(x: int) returns (y: int);
+let
+   mode m1
+   (require ::m2;);
+
+   mode m2
+   (require ::m1;);
+tel
```

Comment:
This test is identical to `tests/lustre/typechecking/shouldFail/test_circular_mode_defs.lus`. The name of this test suggests there should be a circular dependency in the definition of two or more contract equations.

Reply:
changed this to a cyclic dependency for a contract equation.